File: 7_kyu/Odd_or_Even.lua
local solution = {}

-- Returns "odd" if the sum of the numbers in arr is odd, else returns "even".
function solution.odd_or_even(arr)
  local sum = 0
  for _,x in ipairs(arr) do
    sum = ( sum ~ x ) & 1
  end
  if sum ~= 0 then
    return "odd"
  else
    return "even"
  end
end

return solution

File: 7_kyu/Odd_or_Even.rs
// Checks if sum of numbers is even or odd.
fn odd_or_even(numbers: Vec<i32>) -> String {
    if numbers.iter().fold(0i8, |accu, x| accu ^ (x & 1) as i8) == 0 {
        "even"
    } else {
        "odd"
    }
    .to_string()
}
